### ISSUE TITLE
Support dependency-injection

### ISSUE THREAD

**Issue by goofballLogic** [linked-data-dotnet/json-ld.net#66]
The normal usage pattern for the library is to use the static JsonLdProcessor class. This works well for the stateless, static algorithms which make up the JSON-LD standard. However, it requires a consumer to create an injectable facade class in order to inject the APIs into consumers. e.g.

```csharp
interface IJsonLdProcess {
    JObject Compact(JToken input, JToken context, JsonLdOptions opts);
}

class JsonLdProcess {
    JObject Compact(JToken input, JToken context, JsonLdOptions opts) {
        return JsonLdProcessor.Compact(input, context, opts);
    }
}

public class Consumer {

    private readonly JToken _context;
    private readonly IJsonLdProcess _jsonLdProcess;

    public Consumer(IJsonLdProcess jsonLdProcess, JToken context) {
        _jsonLdProcess = jsonLdProcess;
        _context = context;
    }

    public JObject Execute(JObject input) =>
        _jsonLdProcess.Compact(input, _context);
    }
}
```

Can we provide the functionality exposed by JsonLdProcessor as a non-static implementation of an interface which can be mocked for purposes of unit testing / decoration.

**Comment by asbjornu** [comment]
> ```cs
> interface IJsonLdProcess {
>     JObject Compact(JToken input, JToken context, JsonLdOptions opts);
> }
> ```

This interface would require a dependency on Json.NET, so I would prefer a signature that only dealt with objects we defined in our own Core. Something like this:

```cs
interface ILinkedDataProcessor
{
    LinkedDataObject Compact(LinkedDataObject input, LinkedDataContext context, LinkedDataOptions options);
}
```

This would make the Core to be dependency-free and allow Json.NET, `System.Text.Json`, Turtle, and possibly other serializations as ancillary projects to the Core.

**Comment by goofballLogic** [comment]
Agreed. I was not proposing an exact interface here. I also think that we should allow that #66 and #65 might be implemented in either order since we are evolving the codebase
